extensions/resource-deployment/src/services/azdataService.ts

```diff
@@ -0,0 +1,133 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+import * as cp from 'child_process';
+import * as path from 'path';
+import * as fs from 'fs';
+import { IPlatformService } from './platformService';
+
+export interface DeploymentProfile {
+	name: string;
+	defaultDataSize: string;
+	defaultLogSize: string;
+	master: string;
+	data: string;
+	compute: string;
+	hdfs: string;
+	nameNode: string;
+	spark: string;
+	activeDirectory: boolean;
+	hadr: string;
+	includeSpark: string;
+	gatewayPort: string;
+	appProxyPort: string;
+	masterSqlServerPort: string;
+	readableSecondaryPort?: string;
+	controllerPort: string;
+	defaultDataStorageClass: string;
+	defaultLogsStorageClass: string;
+}
+
+interface BdcConfigListOutput {
+	stdout: string[];
+}
+
+export interface IAzdataService {
+	getDeploymentProfiles(): Thenable<DeploymentProfile[]>;
+}
+
+export class AzdataService implements IAzdataService {
+	constructor(private platformService: IPlatformService) {
+	}
+
+	getDeploymentProfiles(): Thenable<DeploymentProfile[]> {
+		return this.getDeploymentProfileNames().then((names: string[]) => {
+			const profilePromises: Thenable<DeploymentProfile>[] = [];
+			names.forEach(name => {
+				profilePromises.push(this.getDeploymentProfileInfo(name));
+			});
+			return Promise.all(profilePromises);
+		});
+	}
+
+	private getDeploymentProfileNames(): Thenable<string[]> {
+		const promise = new Promise<string[]>((resolve, reject) => {
+			cp.exec('azdata bdc config list -o json', (error, stdout, stderror) => {
+				if (error) {
+					reject(error.message);
+				} else {
+					try {
+						const output = <BdcConfigListOutput>JSON.parse(stdout);
+						resolve(output.stdout);
+					}
+					catch (err) {
+						if (err instanceof Error) {
+							reject(err.message);
+						} else {
+							reject(err);
+						}
+					}
+				}
+			});
+		});
+		return promise;
+	}
+
+	private getDeploymentProfileInfo(profileName: string): Thenable<DeploymentProfile> {
+		const promise = new Promise<DeploymentProfile>((resolve, reject) => {
+			if (!fs.existsSync(this.platformService.storagePath())) {
```

Comment:
You should avoid using sync methods, specially in promise based functions anyways.